src/audio/dsm/sof_dsm_inf.c

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2020, Maxim Integrated
+
+#include <sof/audio/sof_dsm.h>
+#include <sof/audio/component.h>
+#include <sof/trace/trace.h>
+#include <sof/drivers/ipc.h>
+#include <sof/ut.h>
+
+#include <sof/audio/channel_map.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <ipc/channel_map.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+static void *_wrap_buf_ptr(void *ptr, const struct audio_stream *buffer)
+{
+	if (ptr >= buffer->end_addr)
+		ptr = (char *)buffer->addr +
+			((char *)ptr - (char *)buffer->end_addr);
+	return ptr;
+}
+
+int sof_dsm_inf_reset(struct sof_dsm_struct_t *hsof_dsm, struct comp_dev *dev)
+{
+	if (!hsof_dsm->dsm_init)
+		return -EINVAL;
+
+	comp_dbg(dev, "[DSM] DSM Reset (handle:%p)", (uintptr_t)hsof_dsm);
+
+	memset(hsof_dsm->buf.sof_a_frame_in, 0,
+	       SOF_FF_BUF_DB_SZ * sizeof(int32_t));
+	memset(hsof_dsm->buf.sof_a_frame_out, 0,
+	       SOF_FF_BUF_DB_SZ * sizeof(int32_t));
+	memset(hsof_dsm->buf.sof_a_frame_iv, 0,
+	       SOF_FB_BUF_DB_SZ * sizeof(int32_t));
+
+	memset(hsof_dsm->buf.stage, 0, DSM_FF_BUF_DB_SZ * sizeof(int32_t));
+	memset(hsof_dsm->buf.stage_fb, 0, DSM_FB_BUF_DB_SZ * sizeof(int32_t));
+
+	memset(hsof_dsm->buf.input, 0, DSM_FF_BUF_SZ * sizeof(int16_t));
+	memset(hsof_dsm->buf.output, 0, DSM_FF_BUF_SZ * sizeof(int16_t));
+	memset(hsof_dsm->buf.voltage, 0, DSM_FF_BUF_SZ * sizeof(int16_t));
+	memset(hsof_dsm->buf.current, 0, DSM_FF_BUF_SZ * sizeof(int16_t));
+
+	memset(hsof_dsm->buf.ff.buf, 0, DSM_FF_BUF_DB_SZ * sizeof(int32_t));
+	memset(hsof_dsm->buf.ff_out.buf, 0, DSM_FF_BUF_DB_SZ * sizeof(int32_t));
+	memset(hsof_dsm->buf.fb.buf, 0, DSM_FB_BUF_DB_SZ * sizeof(int32_t));
+
+	hsof_dsm->seq_ff = 0;
+	hsof_dsm->seq_fb = 0;
+	hsof_dsm->seq_sof_in = 0;
+
+	hsof_dsm->buf.ff.avail = DSM_FF_BUF_SZ;
+	hsof_dsm->buf.ff_out.avail = 0;
+	hsof_dsm->buf.fb.avail = 0;
+
+	return 0;
+}
+
+int sof_dsm_inf_create(struct sof_dsm_struct_t *hsof_dsm, struct comp_dev *dev)
+{
+	return sof_dsm_create(hsof_dsm, dev);
+}
+
+static int sof_dsm_inf_get_buffer(int32_t *buf, uint32_t frames,
+				  struct audio_stream *stream,
+				  int num_channel)
+{
+	int x, y;
+	union sof_dsm_buf input, output;
+
+	input.buf16 = (int16_t *)stream->r_ptr;
+	input.buf32 = (int32_t *)stream->r_ptr;
+	output.buf16 = (int16_t *)buf;
+	output.buf32 = (int32_t *)buf;
+
+	switch (stream->frame_fmt) {
+	case SOF_IPC_FRAME_S16_LE:
+		for (x = 0 ; x < frames ; x++)            {
```

Comment:
I think you have tabs after your parenthesis which is spacing your brackets really weirdly.

Reply:
Thanks. Shall fix it.